whitepaper/wlroots.tex

```diff
@@ -0,0 +1,557 @@
+\maketitle
+
+\begin{abstract}
+wlroots is a flexible and modular library upon which Wayland compositors can be
+built. It is designed to maximize several factors: code reusability,
+interoperability between Wayland compositors, and flexibility for use in novel
+compositor designs.
+\end{abstract}
+
+\newpage
+\tableofcontents
+
+\newpage
+\section{Background}\label{background}
+
+A wlroots compositor has three primary responsibilities.
+
+\begin{enumerate}
+	\item Manage input and output hardware and make them available to desktop
+		applications through the Wayland protocol.
+	\item Manage windows from desktop applications including window screen
+		coordinates, size, state changes (such as maximization and fullscreen),
+		and decorations through the Wayland shell protocols.
+    \item Render the contents of the screen.
+\end{enumerate}
+
+In general terms, to \emph{composite} means to take many seprate images and
+combine them into one image. The screen you are looking at right now is
+presented to you as a single image, but the source of these images may come from
+many different applications or processes. The data for these images (called
+\emph{buffers}) is continuously sent to the compositor through the Wayland
+protocol as the window updates its contents, and the compositor \emph{paints}
+these contents on the screen in the right location.
+
+A Wayland compositor is similar to an X11 window manager. Through the years,
+many novel advancements in desktop UI design have come from these projects. For
+instance, i3wm offers a flexible tree-based tiling model of window placement,
+and projects like Awesome and xmonad have powerful embedded plugin systems.
+wlroots is designed to accomidate projects like these (on any device) as the
+Linux Desktop moves toward Wayland.
+
+wlroots (stylized in lowercase, or as "wlr") is a subproject of the Sway
+compositor. Originally, Sway was written on top of a similar library called wlc.
+This library offered abstractions on top of DRM, GBM, GLES, libinput, and so on
+-- allowing Sway to focus on building the software specific to the compositor.
+However, with time it became clear that wlc's design is too limiting, and many
+of Sway's long-term goals were difficult to achieve based on it.
+
+\subsection{Alternatives}\label{alternatives}
+
+Several alternative approaches to building compositors exist.\\
+
+\textbf{libweston} is the most notable alternative. It is a subproject of the
+Weston compositor, the reference Wayland compositor. However, it is generally
+poorly suited to novel compositors that don't use the traditional desktop
+paradigm. It is also less modular and difficult to break into several pieces.
+Additionally, as a consequence of being refactored out of an existing
+compositor, it inherits many design decisions of Weston itself, which may not be
+amenable to other compositors.
+
+\textbf{swc} and \textbf{wlc} are similar. They both strike a reasonable balance
```

Comment:
`swc` is first seen here and we do not know what it stands for.

Reply:
It's the name of a library, which is why it's bold.

Reply:
Right, but other libraries are introduced in previous text. `libweston` just before and `wlc` before that. If it's a well known library then maybe it isn't necessary, though I think it should be.

(I'm just reading the text and have been following issues here and in sway, I know close to nothing about graphics stuff...)